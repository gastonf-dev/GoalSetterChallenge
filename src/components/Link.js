/**
 * @flow stric-local
 * @format
 */

import React from 'react';
import type {Node} from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import COLORS from '../styles/Colors';

type Props = {
  title: string,
  onPress: function,
};
const Link = ({title, onPress}: Props): Node => {
  return (
    <Text style={styles.link} onPress={onPress} testID="link">
      {title}
    </Text>
  );
};

export default Link;

const styles = StyleSheet.create({
  link: {
    fontFamily: Platform.OS === 'ios' ? 'Avenir LT Std' : 'AvenirLTStd',
    fontSize: 14,
    color: COLORS.lightblue,
  },
});
