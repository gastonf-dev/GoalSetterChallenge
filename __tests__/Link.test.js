import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent, waitFor} from '@testing-library/react-native';

import Link from '../src/components/Link';

describe('<Link />', () => {
  const PROPS = {title: 'Test link', onPress: jest.fn()};
  const wrapper = renderer.create(<Link {...PROPS} />);

  test('render link', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render link and press it', async () => {
    const ON_PRESS_LINK = jest.fn();
    const PROPS = {title: 'Test link', onPress: ON_PRESS_LINK};

    const {getByTestId} = render(<Link {...PROPS} />);
    fireEvent.press(getByTestId('link'));
    await waitFor(() => {
      expect(ON_PRESS_LINK).toHaveBeenCalled();
    });
  });
});
