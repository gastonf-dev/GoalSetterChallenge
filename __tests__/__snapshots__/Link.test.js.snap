// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> render link 1`] = `
<Text
  onPress={[MockFunction]}
  style={
    Object {
      "color": "#17BFDD",
      "fontFamily": "AvenirLTStd",
      "fontSize": 14,
    }
  }
  testID="link"
>
  Test link
</Text>
`;
